File: blind75/php/productofarrayexceptself.php
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer[]
     */
    function productExceptSelf($nums) {
        $length = count($nums);
        $answer = array_fill(0, $length, 1);
        for ($i = 1; $i < $length; $i++) {
            $answer[$i] = $nums[$i - 1] * $answer[$i - 1];
        }
        $R = 1;
        for ($i = $length - 1; $i >= 0; $i--) {
            $answer[$i] = $answer[$i] * $R;
            $R *= $nums[$i];
        }
        return $answer;
    }
}

File: blind75/c++/productofarrayexceptself.cpp
class Solution {
public:
    vector<int> productExceptSelf(vector<int>& nums) {
        int length = nums.size();
        vector<int> answer(length);
        answer[0] = 1;
        for (int i = 1; i < length; i++) {
            answer[i] = nums[i - 1] * answer[i - 1];
        }
        int R = 1;
        for (int i = length - 1; i >= 0; i--) {
            answer[i] = answer[i] * R;
            R *= nums[i];
        }
        return answer;
    }
};
